### ISSUE TITLE
JDK16 Test: TestConstructorFinder fails

### ISSUE THREAD

**Issue by babsingh** [eclipse-openj9/openj9#11872]
### Error details

With OpenJ9 MHs, the error is:

```
java.lang.InternalError: Compile stub invoked! Apart from deliberate reflective access, this should not happen. Please report this to the project so it can be addressed
	at java.base/java.lang.invoke.OpenJDKCompileStub.OpenJDKCompileStubThrowError(OpenJDKCompileStub.java:38)
	at java.base/java.lang.invoke.MethodHandle.intrinsicName(MethodHandle.java:933)
	at java.base/java.lang.invoke.MethodHandleImpl.makeIntrinsic(MethodHandleImpl.java:1444)
	at java.base/java.lang.invoke.MethodHandleImpl$ArrayAccessor.<clinit>(MethodHandleImpl.java:158)
	at java.base/java.lang.Class.forNameImpl(Native Method)
	at java.base/java.lang.Class.forName(Class.java:336)
	at Task.getClasses(Task.java:134)
	at TestConstructorFinder.main(TestConstructorFinder.java:43)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at com.sun.javatest.regtest.agent.MainWrapper$MainThread.run(MainWrapper.java:127)
	at java.base/java.lang.Thread.run(Thread.java:836)
```

With OJDK MHs enabled in OpenJ9, the error is:
```
java.lang.InternalError: java.lang.InternalError: found java.lang.invoke.LambdaForm$DMH/0x0000000000000000, expected java.lang.invoke.MethodHandleImpl$BindCaller
        at java.base/java.lang.invoke.MethodHandleImpl$BindCaller.<clinit>(MethodHandleImpl.java:1233)
        at java.base/java.lang.Class.forNameImpl(Native Method)
        at java.base/java.lang.Class.forName(Class.java:346)
        at Task.getClasses(Task.java:134)
        at TestConstructorFinder.main(TestConstructorFinder.java:43)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:78)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:567)
        at com.sun.javatest.regtest.agent.MainWrapper$MainThread.run(MainWrapper.java:127)
        at java.base/java.lang.Thread.run(Thread.java:853)
Caused by: java.lang.InternalError: found java.lang.invoke.LambdaForm$DMH/0x0000000000000000, expected java.lang.invoke.MethodHandleImpl$BindCaller
        at java.base/java.lang.invoke.MethodHandleImpl$BindCaller.checkCallerClass(MethodHandleImpl.java:1243)
        at java.base/java.lang.invoke.LambdaForm$DMH/0x0000000000000000.invokeStatic(LambdaForm$DMH)
        at java.base/java.lang.invoke.LambdaForm$MH/0x0000000000000000.invokeExact_MT(LambdaForm$MH)
        at java.base/java.lang.invoke.MethodHandleImpl$BindCaller.<clinit>(MethodHandleImpl.java:1231)
        ... 10 more
```

### Run the test

```
# Execute from JDK16 extensions repo
make test TEST="java/beans/XMLDecoder/8028054/TestConstructorFinder.java"

# Using jtreg
jtreg -jdk:<TEST_JDK_HOME> -va test/jdk/java/beans/XMLDecoder/8028054/TestConstructorFinder.java
```

### Resources
- JDK16 extensions repo: https://github.com/ibmruntimes/openj9-openjdk-jdk16
- JTreg-5.1: https://ci.adoptopenjdk.net/view/Dependencies/job/jtreg/lastSuccessfulBuild/artifact/

**Comment by babsingh** [comment]
This should be tagged with `project:MH`.

**Comment by pshipton** [comment]
I won't tag this as jdk16 since with OpenJ9 MH the problem occurs on jdk11 as well, possibly other versions.

**Comment by tajila** [comment]
this looks similar to https://github.com/eclipse/openj9/issues/11433

**Comment by babsingh** [comment]
**Update:** As of 16 Mar 21, this `TestConstructorFinder` passes with OJDK MHs. I verified this using JDK16. `TestConstructorFinder` still fails with OpenJ9 MHs.

**Comment by babsingh** [comment]
The above comment, https://github.com/eclipse/openj9/issues/11872#issuecomment-800609744, reports a false positive. 

I verified that the test passes by running it directly using `jtreg` on my local machine:
```
jtreg/bin/jtreg -jdk:$PWD/build/linux-x86_64-server-release/images/jdk -va test/jdk/java/beans/XMLDecoder/8028054/TestConstructorFinder.java
```
With `jtreg`, the reference implementation and OpenJ9 w/ OJDK MHs have the same output where the test passes.

But, the test still fails with the `make` framework.
```
make test TEST="java/beans/XMLDecoder/8028054/TestConstructorFinder.java"
```
The issue will stay unresolved until the above discrepancy is fixed.

@gacholio @tajila Can you confirm if the following PRs will fix this issue? #12148 and #12194

**Comment by gacholio** [comment]
Again, I don't see anything in there that obviously relates to those PRs.
